commit 28d337cd218cd245b8b4356feb3d3de4522db33a
Author: Luiz Augusto Von Dentz <luiz.dentz-von@nokia.com>
Date:   Wed Feb 17 11:22:16 2010 +0200

    Fix possible crash while verifying avdtp version
    
    Reproducible with Codenomicon A2DP testsuite 1.4.2.

diff --git a/audio/avdtp.c b/audio/avdtp.c
index 3b9c6c1..efed9b6 100644
--- a/audio/avdtp.c
+++ b/audio/avdtp.c
@@ -2125,7 +2125,7 @@ static uint16_t get_version(struct avdtp *session)
 		goto done;
 
 	proto_desc = sdp_get_proto_desc(protos, AVDTP_UUID);
-	if (proto_desc->dtd == SDP_UINT16)
+	if (proto_desc && proto_desc->dtd == SDP_UINT16)
 		ver = proto_desc->val.uint16;
 
 	sdp_list_foreach(protos, (sdp_list_func_t) sdp_list_free, NULL);
